### PR TITLE
Changes for reading data from redis as a defined schema

### PR DESCRIPTION
**Pull request by ishawakankar**

### CODE REVIEW

File: data-products/src/main/scala/org/sunbird/analytics/job/report/AssessmentMetricsJobV2.scala [hunk 1]
```diff
@@ -82,11 +82,11 @@ object AssessmentMetricsJobV2 extends optional.Application with IJob with BaseRe
...
-    spark
-      .read
-      .format(url)
-      .options(settings)
-      .load()
+    val schema = Encoders.product[UserData].schema
+    if(url.equals("org.apache.spark.sql.redis")) { spark.read.schema(schema).format(url).options(settings).load() }
+    else {
+      spark.read.format(url).options(settings).load()
+    }
```
Comment (SanthoshVasabhaktula) [inline data-products/src/main/scala/org/sunbird/analytics/job/report/AssessmentMetricsJobV2.scala:89]: pass the schema as method parameter and register it only if it is not empty

File: data-products/src/main/scala/org/sunbird/analytics/job/report/CourseMetricsJobV2.scala [hunk 2]
```diff
@@ -71,7 +71,11 @@ object CourseMetricsJobV2 extends optional.Application with IJob with ReportGene
 
 // $COVERAGE-ON$ Enabling scoverage for all other functions
   def loadData(spark: SparkSession, settings: Map[String, String], url: String): DataFrame = {
-    spark.read.format(url).options(settings).load()
+    val schema = Encoders.product[UserData].schema
+    if(url.equals("org.apache.spark.sql.redis")) { spark.read.schema(schema).format(url).options(settings).load() }
+    else {
+      spark.read.format(url).options(settings).load()
+    }
```
Comment (SanthoshVasabhaktula) [inline data-products/src/main/scala/org/sunbird/analytics/job/report/CourseMetricsJobV2.scala:78]: Same as earlier comment.

File: data-products/src/main/scala/org/sunbird/analytics/job/report/AssessmentMetricsJobV2.scala (hunk 1 above)
Comment (ishawakankar) [reply data-products/src/main/scala/org/sunbird/analytics/job/report/AssessmentMetricsJobV2.scala]: Done

File: data-products/src/main/scala/org/sunbird/analytics/job/report/CourseMetricsJobV2.scala (hunk 2 above)
Comment (ishawakankar) [reply data-products/src/main/scala/org/sunbird/analytics/job/report/CourseMetricsJobV2.scala]: Done

**Pull request merged**